FlashDevelop/Dialogs/ClipboardHistoryDialog.cs

```diff
@@ -0,0 +1,298 @@
+            preview.DetectUrls = false;
+            preview.Location = new Point(12, 3);
+            preview.Name = "preview";
+            preview.ReadOnly = true;
+            preview.ShortcutsEnabled = false;
+            preview.Size = new Size(458, 227);
+            preview.TabIndex = 2;
+            preview.Text = "";
+            preview.WordWrap = false;
+            // 
+            // ClipboardHistoryDialog
+            // 
+            AcceptButton = btnPaste;
+            AutoScaleDimensions = new SizeF(8f, 20f);
+            AutoScaleMode = AutoScaleMode.Font;
+            CancelButton = btnCancel;
+            ClientSize = new Size(482, 453);
+            Controls.Add(splitContainer);
+            FormBorderStyle = FormBorderStyle.Sizable;
+            MaximizeBox = false;
+            MinimizeBox = false;
+            MinimumSize = new Size(300, 400);
+            Name = "ClipboardHistoryDialog";
+            ShowIcon = false;
+            ShowInTaskbar = false;
+            SizeGripStyle = SizeGripStyle.Show;
+            StartPosition = FormStartPosition.CenterParent;
+            Text = "ClipboardHistoryDialog";
+            splitContainer.Panel1.ResumeLayout(false);
+            splitContainer.Panel2.ResumeLayout(false);
+            splitContainer.ResumeLayout(false);
+            ResumeLayout(false);
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Gets the selected <see cref="ClipboardTextData"/>.
+        /// </summary>
+        public ClipboardTextData SelectedData
+        {
+            get { return (ClipboardTextData) listBox.SelectedItem; }
+        }
+
+        #endregion
+
+        #region Initialization
+
+        private void InitializeFont()
+        {
+            Font = Globals.Settings.DefaultFont;
+            listBox.ItemHeight = Font.Height;
+            preview.Font = Globals.Settings.ConsoleFont;
+        }
+
+        private void InitializeLocalization()
+        {
+            btnPaste.Text = TextHelper.GetStringWithoutMnemonics("Label.Paste");
+            btnCancel.Text = TextHelper.GetStringWithoutMnemonics("Label.Cancel");
+            btnClear.Text = TextHelper.GetStringWithoutMnemonics("Label.Clear");
+        }
+
+        private void InitializeListBox()
+        {
+            listBox.BeginUpdate();
+            foreach (var data in ClipboardManager.History)
+            {
+                listBox.Items.Insert(0, data);
+            }
+            listBox.EndUpdate();
+
+            if (listBox.Items.Count > 0)
+            {
+                listBox.SelectedIndex = 0;
+                btnClear.Enabled = true;
+            }
+        }
+
+        #endregion
+
+        #region Events
+
+        private void ListBox_DrawItem(object sender, DrawItemEventArgs e)
+        {
+            e.DrawBackground();
+
+            string text = (e.Index + 1) + "    " + listBox.GetItemText(listBox.Items[e.Index]).TrimStart();
```

Comment:
Currently the ListBox only displays the first line:
![image](https://cloud.githubusercontent.com/assets/13545633/25619083/8b153966-2f84-11e7-8a75-4b2bf111222d.png)
If I change the code to
```c#
string[] lines = listBox.GetItemText(listBox.Items[e.Index]).Split('\n');
for (int i = 0; i < lines.Length; i++)
{
    lines[i] = lines[i].Trim();
}
string text = (e.Index + 1) + "    " + string.Join(" ", lines);
```
then it merges all lines together:
![image](https://cloud.githubusercontent.com/assets/13545633/25619490/0967681a-2f86-11e7-99ac-852f6728f0db.png)

Which would be better?